fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/conditions/AnyModsLoadedResourceCondition.java

```diff
@@ -0,0 +1,29 @@
+package net.fabricmc.fabric.api.resource.conditions.v1.conditions;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import net.fabricmc.fabric.api.resource.conditions.v1.DefaultResourceConditionTypes;
+import net.fabricmc.fabric.api.resource.conditions.v1.ResourceCondition;
+import net.fabricmc.fabric.api.resource.conditions.v1.ResourceConditionType;
+import net.fabricmc.fabric.impl.resource.conditions.ResourceConditionsImpl;
+
+import net.minecraft.util.dynamic.Codecs;
+
+import java.util.List;
+
+public record AnyModsLoadedResourceCondition(List<String> modIds) implements ResourceCondition {
```

Comment:
We shouldnt expose the record as public API, it makes it much harder to make none breaking changes later. Use a factory method in the .api package, the record is fine to live in .impl.